src/app.py

```diff
@@ -1592,15 +1592,16 @@ def _call_config_script(self, action, env=None):
 
         # Call config script to extract current values
         logger.debug(f"Calling '{action}' action from config script")
-        app_id, app_instance_nb = _parse_app_instance_name(self.entity)
-        settings = _get_app_settings(app_id)
+        app = self.entity
+        app_id, app_instance_nb = _parse_app_instance_name(app)
+        settings = _get_app_settings(app)
```

Comment:
The true fix is this line. The other are just semantic fixes to avoid it happens again.